Lesson17/ecenaz/contactBook.js

```diff
@@ -0,0 +1,235 @@
+  { name: "Dilek", phone: "0 224 555 88 77", email: "xyz@yahoo.com" },
+  { name: "Ali", phone: "0 225 555 88 77", email: "xds@yahoo.com" },
+  { name: "Ayşe", phone: "0 224 555 88 55", email: "sdf@yahoo.com" },
+];
+
+/*
+-----------------------------------------------------------
+  STEP 2: Display All Contacts
+-----------------------------------------------------------
+Function: displayAllContacts()
+- Loops over the 'contacts' array.
+- Logs a descriptive string for each contact.
+
+Example output:
+  Name: Alice, Phone: 123-456-7890, Email: alice@example.com
+*/
+
+function displayAllContacts(contactList, uppercaseMode) {
+  for (const contact of contacts) {
+    if (uppercaseMode) {
+      console.log(
+        `NAME: ${contact.name}, PHONE:${contact.phone}, EMAIL:${contact.email}`,
+      );
+      continue;
+    } else {
+      console.log(
+        `name: ${contact.name}, phone:${contact.phone}, email:${contact.email}`,
+      );
+    }
+  }
+}
+
+console.log("Initial contact list:");
+displayAllContacts(contacts);
+
+/*
+-----------------------------------------------------------
+  STEP 3: Add a New Contact
+-----------------------------------------------------------
+Function: addContact(name, phone, email)
+- Creates a new contact object and pushes it into 'contacts'.
+- Checks if a contact with the same name already 
+  exists before adding. If found, logs a warning and returns.
+- Logs "Contact added successfully." if everything is good.
+*/
+
+function addContact(newName, newPhone, newEmail, contactList) {
+  for (let i = 0; i < contactList.length; i++) {
+    if (newName === contactList[i].name) {
+      console.warn("The contact already exist");
+      return;
+    }
+  }
+  const contactListLengthPrev = contactList.length;
+  contactList.push({
+    name: newName,
+    phone: newPhone,
+    email: newEmail,
+  });
+  const contactListLengthNew = contactList.length;
+
+  if (contactListLengthPrev < contactListLengthNew) {
+    console.log("Contact added successfully.");
+  } else {
+    console.log("An error occured when adding a contact.");
+  }
+
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_operator
+  // contactListLengthPrev < contactListLengthNew
+  //   ? console.log("Contact added successfully.")
+  //   : console.log("An error occured when adding a contact.");
+}
+
+console.log("\nAdding a new contact: Charlie");
+addContact("Charlie", "777-777-7777", "charlie@example.com", contacts);
+displayAllContacts(contacts);
+
+/*
+-----------------------------------------------------------
+  STEP 4: View a Contact by Name
+-----------------------------------------------------------
+Function: viewContact(name)
+- Loops over 'contacts' to find one matching 'name'.
+- Logs the contact info if found.
+- Otherwise, logs: "No contact found with the name: <name>"
+*/
+
+function viewContact(name, contactList) {
+  let isContactFound = false;
+  for (const contact of contactList) {
```

Comment:
The first requirement of this assignment is to refactor this function so that it uses 'findContact' function to check if the contact exists instead of using a loop.